### PR TITLE
Add configurations for increasing gas limit to to 90m per block

### PR DESCRIPTION
**Pull request by henridevieux**
Adds the script and configurations for an increase of the Base mainnet gas limit to 90,000,000 per block.

### CODE REVIEW

File: mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {
+    MultisigBuilder,
+    IMulticall3,
+    IGnosisSafe,
+    console,
+    Enum
+} from "@base-contracts/script/universal/MultisigBuilder.sol";
+
+abstract contract UpdateGaslimit is MultisigBuilder {
```
Comment (stevieraykatz) [inline mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol:13]: We shouldn't mark scripts as `Abstract`. Instead, we should implement all of the virtual methods in the Parent contract. In this case, we're missing `_postCheck()`. See the suggestion below for how to possibly handle this.

File: mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {
+    MultisigBuilder,
+    IMulticall3,
+    IGnosisSafe,
+    console,
+    Enum
+} from "@base-contracts/script/universal/MultisigBuilder.sol";
+
```
Comment (stevieraykatz) [inline mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol:12]:
```suggestion
import { Vm } from "forge-std/Vm.sol";
```

File: mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    function _ownerSafe() internal view override returns (address) {
+        return SYSTEM_CONFIG_OWNER;
+    }
+
+    function _addOverrides(address _safe) internal view override returns (SimulationStateOverride memory) {
+        IGnosisSafe safe = IGnosisSafe(payable(_safe));
+        uint256 _nonce = _getNonce(safe);
+        return overrideSafeThresholdOwnerAndNonce(_safe, DEFAULT_SENDER, _nonce);
+    }
+}
```
Comment (stevieraykatz) [inline mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol:39]:
```suggestion
    }
    
        function _postCheck(Vm.AccountAccess[] memory, SimulationPayload memory) internal override {
        assert(SystemConfig(L1_SYSTEM_CONFIG).gasLimit() == GAS_LIMIT);
    }
}
```

Comment (stevieraykatz) [reply mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol:39]: Using the AccountAccess[] and SimulationPayload fields, we have the flexibility to get more advanced with our script-based validation. In this case though, I think that checking that the config storage is updated successfully is sufficient.

File: mainnet/2024-05-23-increase-gas-limit/script/RollbackGasLimit.sol
Comment (stevieraykatz) [inline mainnet/2024-05-23-increase-gas-limit/script/RollbackGasLimit.sol:1]: See comments on UpdateGasLimit. These same changes can apply here as well.

File: mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol (hunk 1 above)
Comment (henridevieux) [reply mainnet/2024-05-23-increase-gas-limit/script/UpdateGasLimit.sol]: ty!

File: mainnet/2024-05-23-increase-gas-limit/README.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
+# Update Gas Limit in L1 `SystemConfig` 
+
+Status: DONE
+
+## Objective
+
+We are updating the gas limit to improve TPS and reduce gas fees. 
+
+This runbook implements scripts which allow our signers to sign two different calls for our Incident Multisig: 
+1. `UpdateGasLimit` -- This script will update the gas limit to our new limit of 60M gas
```
Comment (0x00101010) [inline mainnet/2024-05-23-increase-gas-limit/README.md:10]: 90M

File: mainnet/2024-05-23-increase-gas-limit/README.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+Status: DONE
+
+## Objective
+
+We are updating the gas limit to improve TPS and reduce gas fees. 
+
+This runbook implements scripts which allow our signers to sign two different calls for our Incident Multisig: 
+1. `UpdateGasLimit` -- This script will update the gas limit to our new limit of 60M gas 
+2. `RollbackGasLimit` -- This script establishes a rollback call in the case we need to revert to 45M gas
```
Comment (0x00101010) [inline mainnet/2024-05-23-increase-gas-limit/README.md:11]: 60M

File: mainnet/2024-05-23-increase-gas-limit/README.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+   you will need to determine which “number” it is in the list of
+   addresses on your ledger.
+4. "Success" with a green check mark 
+
+
+#### 3.2. Validate correctness of the state diff.
+
+Now click on the "State" tab. Verify that:
+
+1. Verify that the nonce is incremented for the Incident Multisig under the "GnosisSafeProxy" at address `0x14536667Cd30e52C0b458BaACcB9faDA7046E056`. We should see the nonce increment from 14 to 15:
```
Comment (0x00101010) [inline mainnet/2024-05-23-increase-gas-limit/README.md:77]: 15 to 16

File: mainnet/2024-05-23-increase-gas-limit/README.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+
+```
+Key: 0x0000000000000000000000000000000000000000000000000000000000000005
+Before: 0x000000000000000000000000000000000000000000000000000000000000000e
+After: 0x0000000000000000000000000000000000000000000000000000000000000000f
+```
+
+2. Verify that gas limit value is appropriately updated under "Proxy" at address `0x73a79fab69143498ed3712e519a88a918e1f4072`. We should see that the gas limit has been changed from 30M to 45M:
```
Comment (0x00101010) [inline mainnet/2024-05-23-increase-gas-limit/README.md:85]: 60M to 90M

File: mainnet/2024-05-23-increase-gas-limit/README.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+
+Status: DONE
+
+## Objective
+
+We are updating the gas limit to improve TPS and reduce gas fees. 
+
+This runbook implements scripts which allow our signers to sign two different calls for our Incident Multisig: 
+1. `UpdateGasLimit` -- This script will update the gas limit to our new limit of 90M gas
+2. `RollbackGasLimit` -- This script establishes a rollback call in the case we need to revert to 90M gas
```
Comment (0x00101010) [inline mainnet/2024-05-23-increase-gas-limit/README.md:11]: nit: 60m

**Review by 0x00101010 — APPROVED** [review]

**Review by stevieraykatz — APPROVED** [review]

**Pull request merged**